libs/wingc/src/parser.rs

```diff
@@ -584,6 +584,93 @@ impl<'s> Parser<'s> {
 		}))
 	}
 
+	fn build_interface_statement(&self, statement_node: &Node) -> DiagnosticResult<StmtKind> {
+		let mut cursor = statement_node.walk();
+		let mut extends = vec![];
+		let mut methods = vec![];
+		let name = self.node_symbol(&statement_node.child_by_field_name("name").unwrap())?;
+
+		for interface_element in statement_node
+			.child_by_field_name("implementation")
+			.unwrap()
+			.named_children(&mut cursor)
+		{
+			if interface_element.is_extra() {
+				continue;
+			}
+			match interface_element.kind() {
+				"method_signature" => {
+					let method_name = self.node_symbol(&interface_element.child_by_field_name("name").unwrap());
+					let func_sig = self.build_function_signature(&interface_element, Phase::Preflight);
+					match (method_name, func_sig) {
+						(Ok(method_name), Ok(func_sig)) => methods.push((method_name, func_sig)),
+						_ => {}
+					}
+				}
+				"inflight_method_signature" => {
+					let method_name = self.node_symbol(&interface_element.child_by_field_name("name").unwrap());
+					let func_sig = self.build_function_signature(&interface_element, Phase::Inflight);
+					match (method_name, func_sig) {
+						(Ok(method_name), Ok(func_sig)) => methods.push((method_name, func_sig)),
+						_ => {}
+					}
+				}
+				"ERROR" => {
+					self
+						.add_error::<Node>(format!("Expected interface element node"), &interface_element)
+						.err();
+				}
+				other => {
+					panic!(
+						"Unexpected interface element node type {} || {:#?}",
+						other, interface_element
+					);
+				}
+			}
+		}
+
+		for extend in statement_node.children_by_field_name("extends", &mut cursor) {
+			// ignore comments
+			if extend.is_extra() {
+				continue;
+			}
+
+			// ignore commas
+			if !extend.is_named() {
+				continue;
+			}
+
+			let interface_type = self.build_udt_annotation(&extend)?;
+			match interface_type {
+				TypeAnnotation::UserDefined(interface_type) => extends.push(interface_type),
+				_ => {
```

Comment:
`build_udt_annotation` alaways returns a udt or an error, so there's no need to handle this, we should just panic.